roles/openshift_health_checker/library/aos_version.py

```diff
@@ -27,14 +28,18 @@ def bail(error):  # pylint: disable=missing-docstring
         module.fail_json(msg=error)
 
     rpm_prefix = module.params['prefix']
-
     if not rpm_prefix:
         bail("prefix must not be empty")
 
-    yb = yum.YumBase()  # pylint: disable=invalid-name
-    yb.conf.disable_excludes = ["all"]  # assume the openshift excluder will be managed, ignore current state
+    # determine what level of precision the user specified for the openshift version.
+    # should look like a version string with possibly many segments e.g. "3.4.1":
+    expected_version = module.params['version']
+    if not re.match(r'\A\d+(?:\.\d+)+\Z', expected_version):
+        bail('Requested release version "%s" does not look like a version.' % expected_version)
 
     # search for package versions available for aos pkgs
+    yb = yum.YumBase()  # pylint: disable=invalid-name
+    yb.conf.disable_excludes = ["all"]  # assume the openshift excluder will be managed, ignore current state
```

Comment:
Do we need to talk about "openshift excluder" here? I fear it may cause more confusion than help.